stl/inc/xutility

```diff
@@ -4667,6 +4667,63 @@ _NODISCARD bool lexicographical_compare(
 }
 #endif // _HAS_CXX17
 
+#if defined(__cpp_impl_three_way_comparison) && defined(__cpp_lib_concepts)
+// FUNCTION TEMPLATE lexicographical_compare_three_way
+template <class _InIt1, class _InIt2, class _Cmp>
+constexpr auto lexicographical_compare_three_way(_InIt1 _First1, _InIt1 _Last1, _InIt2 _First2, _InIt2 _Last2,
+    _Cmp _Comp) -> decltype(_Comp(*_First1, *_First2)) {
+    _Adl_verify_range(_First1, _Last1);
+    _Adl_verify_range(_First2, _Last2);
+    auto _UFirst1      = _Get_unwrapped(_First1);
+    const auto _ULast1 = _Get_unwrapped(_Last1);
+    auto _UFirst2      = _Get_unwrapped(_First2);
+    const auto _ULast2 = _Get_unwrapped(_Last2);
+    using _UIt1        = decltype(_UFirst1);
+    using _UIt2        = decltype(_UFirst2);
+
+    if constexpr (conjunction_v<is_same<_Cmp, compare_three_way>, is_pointer<_UIt1>, is_pointer<_UIt2>,
+                      disjunction<
+                          conjunction<is_same<remove_pointer_t<_UIt1>, byte>, is_same<remove_pointer_t<_UIt2>, byte>>,

```

Comment:
We provide an escape hatch that allows users to remove `std::byte` due to an unfortunate conflict with the Windows SDK (see #204) so this should be guarded. We should test `__cpp_lib_byte` (I have filed #564 because the rest of the STL isn't testing the Standard feature-test macro for this purpose). Otherwise, this header will fail to compile if the user has removed `std::byte`.

Reply:
Done